### PR TITLE
[api][SIMS-#518] Open Award Eligibility in Camunda (CSG- FT) -> updated query and response

### PR DESCRIPTION
**Pull request by ann-aot**
relation data in the application is not available at every point so, added  `null` check in the response and updated query,
eg, 
image
at this point, the application won't be having offering, location info

### CODE REVIEW

File: sources/packages/api/src/route-controllers/application/application.system.controller.ts [hunk 1]
```diff
@@ -53,34 +53,36 @@ export class ApplicationSystemController {
...
+          application.location?.institution?.institutionType?.name,
       },
       location: {
-        institutionLocationProvince: application.location.data.address.province,
+        institutionLocationProvince:
+          application.location?.data?.address?.province,
       },
       student: {
-        studentPDStatus: application.student.studentPDVerified,
+        studentPDStatus: application.student?.studentPDVerified,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/application/application.system.controller.ts:85]: Student will never be nullable.

File: sources/packages/api/src/route-controllers/application/application.system.controller.ts [hunk 2]
```diff
@@ -53,34 +53,36 @@ export class ApplicationSystemController {
 
     return {
       data: application.data,
-      programYear: application.programYear.programYear,
+      programYear: application.programYear?.programYear,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/application/application.system.controller.ts:56]: Program year will never be nullable.

File: sources/packages/api/src/services/application/application.service.ts [hunk 3]
```diff
@@ -343,17 +343,17 @@ export class ApplicationService extends RecordDataModelService<Application> {
         "institutionType",
         "student",
       ])
-      .innerJoin("application.programYear", "programYear")
-      .innerJoin("application.offering", "offering")
-      .innerJoin("application.pirProgram", "pirProgram")
-      .innerJoin("application.location", "location")
+      .leftJoin("application.programYear", "programYear")
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/application/application.service.ts:346]: Program year should be an inner join. An application should not even start the assessment without it.

File: sources/packages/api/src/services/application/application.service.ts [hunk 4]
```diff
@@ -343,17 +343,17 @@ export class ApplicationService extends RecordDataModelService<Application> {
...
+      .leftJoin("application.offering", "offering")
+      .leftJoin("application.pirProgram", "pirProgram")
+      .leftJoin("application.location", "location")
       .leftJoin("location.institution", "institution")
       .leftJoin("institution.institutionType", "institutionType")
-      .innerJoin("application.student", "student")
+      .leftJoin("application.student", "student")
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/application/application.service.ts:352]: Student should be an inner join.

**Review by andrewsignori-aot — COMMENTED** [review]
Please take a look at the comments. 
IMO, student and program year should never be null. An application should not be created without them.

**Review by andrewsignori-aot — APPROVED** [review]
looks good 👍

**Review by guru-aot — APPROVED** [review]
LGTM

**Pull request merged**